pytorch_toolkit/light-weight-face-anti-spoofing/models/mobilenetv2.py

```diff
@@ -110,85 +62,46 @@ def forward(self, x):
             return self.dropout2d(self.conv(x))
 
 
-class MobileNetV2(nn.Module):
-    def __init__(self, width_mult=1., prob_dropout=0.1, type_dropout='bernoulli',
-                 prob_dropout_linear=0.5, embeding_dim=1280, mu=0.5, sigma=0.3,
-                 theta=0, multi_heads=True, scaling=1):
-        super().__init__()
+class MobileNetV2(MobileNet):
+    def __init__(self, cfgs, **kwargs):
+        super().__init__(**kwargs)
         # setting of inverted residual blocks
-        self.multi_heads = multi_heads
-        self.scaling = scaling
-        self.prob_dropout_linear = prob_dropout_linear
-        self.cfgs = [
-            # t, c, n, s
-            [1,  16, 1, 1],
-            [6,  24, 2, 2],
-            [6,  32, 3, 2],
-            [6,  64, 4, 2],
-            [6,  96, 3, 1],
-            [6, 160, 3, 2],
-            [6, 320, 1, 1],
-        ]
+        self.cfgs = cfgs
 
         # building first layer
-        input_channel = _make_divisible(32 * width_mult, 4 if width_mult == 0.1 else 8)
-        layers = [conv_3x3_bn(3, input_channel, 2, theta=theta)]
+        input_channel = make_divisible(32 * self.width_mult, 4 if self.width_mult == 0.1 else 8)
+        layers = [conv_3x3_bn(3, input_channel, 2, theta=self.theta)]
         # building inverted residual blocks
         block = InvertedResidual
         for t, c, n, s in self.cfgs:
-            output_channel = _make_divisible(c * width_mult, 4 if width_mult == 0.1 else 8)
+            output_channel = make_divisible(c * self.width_mult, 4 if self.width_mult == 0.1 else 8)
             for i in range(n):
                 layers.append(block(input_channel, output_channel,
                                     s if i == 0 else 1, t,
-                                    prob_dropout=prob_dropout,
-                                    type_dropout=type_dropout,
-                                    mu=mu, sigma=sigma, theta=theta))
+                                    prob_dropout=self.prob_dropout,
+                                    type_dropout=self.type_dropout,
+                                    mu=self.mu, sigma=self.sigma, theta=self.theta))
                 input_channel = output_channel
         self.features = nn.Sequential(*layers)
-        # building last several layers
-        self.conv_last = conv_1x1_bn(input_channel, embeding_dim)
-        self.avgpool = nn.AdaptiveAvgPool2d((1, 1))
-        self.spoofer = nn.Linear(embeding_dim, 2)
-        if self.multi_heads:
-            self.lightning = nn.Linear(embeding_dim, 5)
-            self.spoof_type = nn.Linear(embeding_dim, 11)
-            self.real_atr = nn.Linear(embeding_dim, 40)
+        self.conv_last = conv_1x1_bn(input_channel, self.embeding_dim)
 
     def forward(self, x):
-        x = self.features(x)
-        x = self.conv_last(x)
+        x = super().make_features(x)
```

Comment:
`forward()` methods are identical between MN-V2,-V3. So move it to the parent class too.